### PR TITLE
Adding RulerZ as a sample for Specification pattern

### PR DESCRIPTION
**Pull request by jhuet**
RulerZ is a PHP implementation of the Specification pattern which puts the emphasis on three main aspects:

  - an easy and data-agnostic DSL to define business rules and specifications,
  - the ability to check if a candidate satisfies a specification,
  - the ability to filter or query any datasource to only retrieve candidates matching a specification.

### CODE REVIEW

File: Behavioral/Specification/README.rst [hunk 1]
```diff
@@ -9,6 +9,11 @@ checked against. The composite specification class has one method called
 ``isSatisfiedBy`` that returns either true or false depending on whether
 the given object satisfies the specification.
 
+Sample:
```
Comment (domnikl) [inline Behavioral/Specification/README.rst]: Right now we also need the modifications in README.md, could you please add it there, too?

**Comment by shangguokan** [comment]
@domnikl there is no .md anymore for every pattern, i just keep REDAME.md Behavioral/REDAME.md Creational/REDAME.md More/REDAME.md and Structual/REDAME.md
if you think RulerZ is a good example for Specification pattern, this commit is ok to merge.

**Comment by domnikl** [comment]
@jhuet sorry, my bad! I am merging it.
@shangguokan thanks for the hint!

**Comment by jhuet** [comment]
@domnikl @shangguokan thank you !

**Pull request merged**